### PR TITLE
CP-10605: Migration of vGPU enabled VMs

### PR DESCRIPTION
**Pull request by ravippandey**

### CODE REVIEW

File: xc/device.ml [hunk 1]
```diff
@@ -1832,6 +1832,9 @@ let vnconly_cmdline ~info ?(extras=[]) domid =
     @ (List.fold_left (fun l (k, v) -> ("-" ^ k) :: (match v with None -> l | Some v -> v :: l)) [] extras)
 
 let vgpu_args_of_info info domid =
+	let path = sprintf "/var/lib/xen/demu-%s.%d" in
+	let suspend_file = path "save" domid in
+	let resume_file = path "resume" domid in
```
Comment (johnelse) [inline xc/device.ml]: This code could re-use `Device_common.demu_save_path` and `Device_common.demu_restore_path`.

File: xc/xenops_server_xen.ml [hunk 2]
```diff
@@ -1228,11 +1228,22 @@ module VM = struct
...
+				let vgpu_guest = Device.Vgpu.is_running ~xs domid in
+
+				(* Suspend dvpgu *)
+				if vgpu_guest then
+				begin
+					debug "This is a vgpu guest";
+					match Device.Vgpu.pid ~xs domid with
+					| None -> debug "No vgpu process running underneath";
+					| Some pid -> debug "Vgpu process suspended";
+					              Unix.kill pid Sys.sighup
```
Comment (johnelse) [inline xc/xenops_server_xen.ml]: There are mixed tabs and spaces in this line.

**Comment by johnelse** [comment]
Other than the two comments this looks good to me.

**Comment by ravippandey** [comment]
Hi John,

I have put the review as an additional commit.

Thanks,
Ravi

**Comment by johnelse** [comment]
Looks good :) I'll merge this once the build has finished.

**Comment by ravippandey** [comment]
Thanks :)

**Pull request merged**